src/provider/provider_os_memory_linux.c

```diff
@@ -80,7 +80,8 @@ static int syscall_memfd_secret(void) {
     int fd = -1;
 #ifdef __NR_memfd_secret
     // SYS_memfd_secret is supported since Linux 5.14
-    fd = syscall(SYS_memfd_secret, 0);
+    // not using SYS_memfd_secret as SLES does not define it
+    fd = syscall(__NR_memfd_secret, 0);
```

Comment:
The same change is probably needed for `SYS_memfd_create` below (line 99)

Reply:
Ok, but still, if we're checking if `__NR_memfd_create` is defined I don't see a reason why we don't use `__NR_memfd_create` in the syscall as well.

Reply:
I agree - @lslusarczyk could you also add changes for memfd_create?

Reply:
@igchor @bratpiorka This issue has been fixed.